node/node.go

```diff
@@ -1329,7 +1340,17 @@ func (node *AlgorandFullNode) AssembleBlock(round basics.Round) (agreement.Valid
 		}
 		return nil, err
 	}
-	return validatedBlock{vb: lvb}, nil
+
+	// ensure UnfinishedBlock contains provided addresses
+	for _, addr := range addrs {
+		if !ub.ContainsAddress(addr) {
+			// this should not happen: VotingKeys() and VotingAccountsForRound() should be in sync
+			node.log.Errorf("AlgorandFullNode.AssembleBlock: could not generate a proposal for round %d, proposer %s not in UnfinishedBlock", round, addr)
+			return nil, agreement.ErrAssembleBlockRoundStale
```

Comment:
does not look a stale block error, need a new error value

Reply:
I'd like @cce to weigh in.  I _ think_ this is still a "staleness" thing. Something about making the block before all the participants are known. So the block is "stale".

Reply:
I might have misunderstood, but it sounds like you're saying you want this logged, but changing the error type would log it twice. So can we make a new error type, remove the new `node.log.Errorf` here, and let `makeProposals` log it higher up, since it will no longer be an `ErrAssembledBlockRoundStale`?

Reply:
How about this slightly different different plan.  Remove all Errorf logging in `func (node *AlgorandFullNode) AssembleBlock`. To my mind, this handling in `makeProposals` is just using the Stale error as a flag to mean "this has already been logged". 
```
 	if err != nil {
		if err != ErrAssembleBlockRoundStale {
			n.log.Errorf("pseudonode.makeProposals: could not generate a proposal for round %d: %v", round, err)
		}
		return nil, nil
	}
```
So I would say, let's _never_ log in `AssembleBlock` and instead annotate the error for the two special cases there using `fmt.Errorf("%w")` and return it, where it will get logged.

Reply:
Ugh, that's no good either since the transaction pool has all sort of different logging for a StaleBlock error. It really does seem like it's being used solely as a signal to mean "Logging has already be handled", but the logging that already happened might be have been info, warn, or even just swallowed.
So I advocate leaving this as is. If this was a common pattern, maybe we'd introduce a `basics.LoggedError` wrapper or something as a signal. But that seems overkill. And this is _kind of_ a stale block.
@algorandskiy is that ok with you?

Reply:
I'm good with making no changes at this point.